src/bijectors/logit.jl

```diff
@@ -26,18 +28,12 @@ up1(b::Logit{N, T}) where {N, T} = Logit{N + 1, T}(b.a, b.b)
 Base.:(==)(b1::Logit, b2::Logit) = b1.a == b2.a && b1.b == b2.b
 
 (b::Logit{0})(x::Real) = _logit(x, b.a, b.b)
-(b::Logit{0})(x) = _logit.(x, b.a, b.b)
-(b::Logit{1})(x::AbstractVector) = _logit.(x, b.a, b.b)
-(b::Logit{1})(x::AbstractMatrix) = _logit.(x, b.a, b.b)
-(b::Logit{2})(x::AbstractMatrix) = _logit.(x, b.a, b.b)
-(b::Logit{2})(x::AbstractArray{<:AbstractMatrix}) = map(b, x)
-_logit(x, a, b) = logit((x - a) / (b - a))
+(b::Logit)(x) = _logit.(x, b.a, b.b)
```

Comment:
Is the broadcasting overhead too large to remove the `(::Logit{0})(x::Real)` definition above? Speaking of overhead, if `x`, `b.a` and `b.b` are of the same dimension, one could also just use `map` which does not involve the huge broadcasting machinery.

Reply:
I haven't checked recently, but I did go through the process of checking the overhead of broadcasting a year or so ago and it was sufficient for me to specialize on `Real`.

EDIT: Just checked and it doens't actually matter anymore (on Julia 1.5.3)! Seems like there isn't any overhead for `Real` these days, so it can prob be removed:)

Reply:
Oh and regarding `map` vs. broadcasting; is there a particular reason why we want to avoid broadcasting?

Reply:
The main reason would be that broadcasting is quite complex and hence  often creates additional overhead and problems. E.g., it sometimes lead to type inference problems and the adjoints in Tracker and Zygote are based on ForwardDiff.

Reply:
Isn't ForwardDiff blazingly fast for broadcasting though, due to it's vectorized mode or w/e?

Reply:
Maybe, I don't know the details there. I just noticed that the implementation is quite non-trivial and still does not work in all cases (IIRC we even noticed some bugs in Bijectors that were caused by this implementation in Tracker, AFAIK at least these problems are fixed now).